cmd/geth/misccmd.go

```diff
@@ -35,7 +35,7 @@ var (
 		Action:    utils.MigrateFlags(makecache),
 		Name:      "makecache",
 		Usage:     "Generate ethash verification cache (for testing)",
-		ArgsUsage: "<blockNum> <outputDir>",
+		ArgsUsage: "<blockNum> <epochLength> <outputDir>",
```

Comment:
I don't love adding a required parameter to a existing commands (since this breaks an API). Maybe instead this can accept `--<chain>` flags to have chain configuration (ie ECIP1099) context, or use a flag `--epoch.length`.

Reply:
Generally speaking I would agree, however in this case, "Generate ethash verification cache **(for testing)**". I feel it's safe to make changes. This doesnt touch chains/params it only touches ethash, generating cache/dag to outputdir and exiting.

Reply:
Optional and defaulting to `epochLengthDefault` makes sense imo.

Reply:
I hear ya... but an API for testing is still an API... (and since it's useable and observable, it may be used for anything.. like miners experimenting with distributed DAGs or operators using it to generate a shared (or redundant) resource ahead of time for a pool using a newfangled microservices infrastructure..)

Is there a way we can make it an optional parameter, while maintaining established default behavior?

Reply:
The complexity here is everywhere else uses ECIP1099FBlock defined in chainconfig, but makecache and makedag, are completely unaware of blocks or chainconfigs. So it needs to somehow be passed the param. Would an `--epoch.length` flag override epochLengthDefault/ECIP1099FBlock in the rest of the logic chain/ethash logic? or does it only apply to `geth makecache` and `geth makedag`?

Reply:
> Would an --epoch.length flag override epochLengthDefault/ECIP1099FBlock in the rest of the chain/ethash logic? or does it only apply to geth makecache and geth makedag?

My thinking was that the flag would only apply to these `makedag` and `makecache` commands (and not show up at all in `geth --help`)...